lm75bd/lm75bd.c

```diff
@@ -27,6 +27,31 @@ error_code_t lm75bdInit(lm75bd_config_t *config) {
 
 error_code_t readTempLM75BD(uint8_t devAddr, float *temp) {
   /* Implement this driver function */
+  error_code_t errCode;
+  uint8_t buff[2] = {0x0};
+  uint16_t signMask = 0x0400; // The 11th bit from right (aka bit 10) in our raw int will represent the sign
+  uint16_t elevenBitMask = 0x07ff; // To do bitwise operations to basically treat a uint16_t like it only has 11 bits
+
+  RETURN_IF_ERROR_CODE(i2cSendTo(devAddr, buff, 1)); // Set pointer register for temp read
+  RETURN_IF_ERROR_CODE(i2cReceiveFrom(devAddr, buff, 2)); // Recieve 2 temperature bytes
+
+  // Collect 11 meaningful bits into a 16-bit integer
+  uint16_t raw = buff[0];
+  raw <<= 3; 
+  raw |= buff[1] >> 5;
+
+  // Assign a value to the provided float, dependent on sign
+  if (raw & signMask) {
+    // Two's complement on raw -> positive representation
+    raw = raw ^ elevenBitMask;
+    raw += 1;
+
+    raw &= elevenBitMask; // Ignore any potential overflow to the 12th bit from right
+
+    *temp = -(raw) * 0.125f;
+  } else {
+    *temp = (raw) * 0.125f;
+  }
```

Comment:
U don't need this just use int16_t to store the combination of data from the buffer. Then multiply by 0.125 after performing the necessary bit shifting